ballista/rust/core/src/execution_plans/query_stage.rs

```diff
@@ -31,26 +45,44 @@ use uuid::Uuid;
 #[derive(Debug, Clone)]
 pub struct QueryStageExec {
     /// Unique ID for the job (query) that this stage is a part of
-    pub job_id: String,
+    job_id: String,
     /// Unique query stage ID within the job
-    pub stage_id: usize,
+    stage_id: usize,
     /// Physical execution plan for this query stage
-    pub child: Arc<dyn ExecutionPlan>,
+    plan: Arc<dyn ExecutionPlan>,
+    /// Path to write output streams to
+    work_dir: String,
+    /// Optional shuffle output partitioning
+    shuffle_output_partitioning: Option<Partitioning>,
 }
 
 impl QueryStageExec {
     /// Create a new query stage
     pub fn try_new(
-        job_id: String,
+        job_id: &str,
```

Comment:
FWIW I think for "constructor methods" it is OK to take an owned `String` as the `&str` will be copied anyway.

Reply:
FWIW another pattern that I like is  `job_id: impl Into<String>` so that the caller can pass in an owned String if they have one or a &str if they don't (or anything else that knows how to turn itself into a String)